src/gui/fspathedit_p.cpp

```diff
@@ -260,17 +207,18 @@ QWidget *Private::FileLineEdit::widget()
 void Private::FileLineEdit::keyPressEvent(QKeyEvent *e)
 {
     QLineEdit::keyPressEvent(e);
+
     if ((e->key() == Qt::Key_Space) && (e->modifiers() == Qt::CTRL))
     {
-        m_completerModel->setRootPath(QFileInfo(text()).absoluteDir().absolutePath());
+        m_completerModel->setRootPath(Path(text()).data());
```

Comment:
I wonder how does completer behave in case of relative path.
FYI, many relative paths in qBittorrent are supposed to be resolved against base path other than application current directory (and in some cases, base path is even unknown until the moment when it is necessary to resolve the relative path, i.e. it depends on some other factors or in other words the path is a template).

Reply:
The completer doesn't recognize relative path. It will show "no path exists" error.
I also tested, using relative path here will be relative to default download path.

IMO relative path isn't user-friendly in current form. GUI doesn't show what is the base path which will relative to.
And there is the problem of portable profile, which users will expect configs/logs/etc. to be stored within the portable profile folder. In this scenario, users (at least me) would expect relative path is relative to the portable profile folder.

Reply:
> I also tested, using relative path at "add torrent download" dialog will be relative to default download path.

Yes. It is designed to behave in this way.


> IMO relative path isn't user-friendly in current form.

In what form?
Relative paths are used to provide some powerful features and automatizations. IMO, it is clearly expected that different kind of paths use different base paths for resolving relative paths, otherwise the mentioned above features aren't impossible to implement.


> GUI doesn't show what is the base path which will relative to.

It is problem of GUI.

> users will expect configs/logs/etc. to be stored within the portable profile folder. In this scenario, users (at least me) would expect relative path is relative to the portable profile folder path.

I don't see any obstacles to implement it. If the relative paths of the config/log files, etc. are resolved relative to the base path of the profile, then this is quite logical and does not contradict what I said above.

> The completer doesn't recognize relative path. It will show "no path exists" error.

Suggestion for further improvement:
Path edit widget (and path validator inside it) can have a property for base path to resolve relative paths.

Reply:
>In what form?
Relative paths are used to provide some powerful features and automatizations. IMO, it is clearly expected that different kind of paths use different base paths for resolving relative paths, otherwise the mentioned above features aren't impossible to implement.

I'm not saying relative path is unwanted/bad. I'm emphasizing how to properly present it in the UI to the user. Without proper UI hints it will be confusing/inaccessible to the users.

>It is problem of GUI.

Yes and it is a big problem IMO. Perhaps adding some descriptions (via tooltips) about how relative path would work for the respective widget will help?

>Suggestion for further improvement:
Path edit widget (and path validator inside it) can have a property for base path to resolve relative paths.

No problem but I had other things on my plate so I probably won't be coding it soon.

Reply:
> Perhaps adding some descriptions (via tooltips) about how relative path would work for the respective widget will help?

Yes, of course.